src/domain/file/services/file-service.ts

```diff
@@ -199,45 +216,59 @@ export class FileService {
                 ACL: 'private',
             })
             .promise();
+
         let partNumber = 1;
         // eslint-disable-next-line @typescript-eslint/no-explicit-any
         const chunks: Array<any> = [];
         const parts: { ETag: string | undefined; PartNumber: number }[] = [];
 
-        const uploadStatusFunction = async (
-            userId: string,
-            filename: string,
-            fileId: string,
-            percentage: number,
-        ): Promise<void> =>
-            await pubsub.publish('UPLOAD_STATUS', {
-                fileUploadProgress: {
-                    userId,
-                    filename,
-                    fileId,
-                    percentage,
-                    type: FileType.MANUSCRIPT_SOURCE,
-                    submissionId,
-                },
-            });
-
+        // tracks bytes until 5mb or last chunk, and send up.
+        let currentBytes = 0;
+        let chunksToSend = [];
         for await (const chunk of stream) {
             const bytesRead = stream.bytesRead;
-            const { ETag } = await this.handleMultipartChunk(
-                userId,
-                file,
-                chunk,
-                partNumber,
-                fileUploadManager,
-                bytesRead,
-                uploadStatusFunction,
-            );
-            parts.push({ ETag, PartNumber: partNumber });
+            currentBytes = currentBytes + chunk.length;
+            chunksToSend.push(chunk);
             chunks.push(chunk);
-            partNumber++;
+            if (currentBytes >= s3MinChunkSize || bytesRead === file.size) {
```

Comment:
Do we check / handle the fact that file.size might not be the same as the total chunk sizes? ... Even if its just for logging purposes - after the for loop exits, check that we've sent the expected size?

Reply:
This should be handled in the higher level middleware for apollo.